cpu/lpc11u34/periph/gpio.c

```diff
@@ -0,0 +1,503 @@
+    }
+    else {
+        LPC_GPIO->DIR[port] &= ~(1 << pin);           /* set pin to output mode */
+
+    }
+
+    return 0; /* all OK */
+}
+
+int gpio_init_int(gpio_t dev, gpio_pp_t pullup, gpio_flank_t flank, gpio_cb_t cb, void *arg)
+{
+    int res;
+    uint8_t pin;
+    uint8_t port;
+
+    LPC_SYSCON->SYSAHBCLKCTRL |= (1 << 19);
+
+    if (dev >= GPIO_NUMOF) {
+        return -1;
+    }
+
+    if (gpio_int_id > 7)
+        return -1;
+
+    port = gpio_port_map[dev];
+    pin = gpio_pin_map[dev];
+
+    /* configure pin as input */
+    res = gpio_init(dev, GPIO_DIR_IN, pullup);
+    if (res < 0) {
+        return res;
+    }
+
+    /* Disable interrupts */
+    NVIC_DisableIRQ(gpio_int_id);
+    /* Register pin as flex interrupt */
+    LPC_SYSCON->PINTSEL[gpio_int_id] = (port * 24) + pin;
+
+    /* set callback */
+    gpio_config[gpio_int_id].cb = cb;
+    gpio_config[gpio_int_id].arg = arg;
+
+    /* configure the event that triggers an interrupt */
+    switch (flank) {
+        case GPIO_RISING:
+            LPC_GPIO_PIN_INT->ISEL &= ~(1 << gpio_int_id); // Set PMODE=edge sensitive
+            LPC_GPIO_PIN_INT->IENR |= (1 << gpio_int_id);  // Enable rising edge.
+            LPC_GPIO_PIN_INT->IENF &= ~(1 << gpio_int_id); // Disable falling edge.
+            break;
+        case GPIO_FALLING:
+            LPC_GPIO_PIN_INT->ISEL &= ~(1 << gpio_int_id); // Set PMODE=edge sensitive
+            LPC_GPIO_PIN_INT->IENR &= ~(1 << gpio_int_id); // Disable rising edge.
+            LPC_GPIO_PIN_INT->IENF |= (1 << gpio_int_id);  // Enable falling edge.
+            break;
+        case GPIO_BOTH:
+            LPC_GPIO_PIN_INT->ISEL &= ~(1 << gpio_int_id); // Set PMODE=edge sensitive
+            LPC_GPIO_PIN_INT->IENR |= (1 << gpio_int_id);  // Enable rising edge.
+            LPC_GPIO_PIN_INT->IENF |= (1 << gpio_int_id);  // Enable falling edge.
+            break;
+    }
+
+    /* clear any pending requests */
+    LPC_GPIO_PIN_INT->RISE = (1 << gpio_int_id); // Clear rising edge (sort of) flag
+    LPC_GPIO_PIN_INT->FALL = (1 << gpio_int_id); // Clear falling edge (sort of) flag
+
+    NVIC_SetPriority(gpio_int_id, 3);
+
+    flex_int_mapping[dev] = gpio_int_id+1;
+
+    gpio_int_id++;
+
+    return 0;
+}
+
+void gpio_irq_enable(gpio_t dev)
+{
+    if (dev >= GPIO_NUMOF) {
+        return;
+    }
+
+    NVIC_EnableIRQ(flex_int_mapping[dev]-1);
+}
+
+void gpio_irq_disable(gpio_t dev)
+{
+    uint8_t int_id;
+
+    if (dev >= GPIO_NUMOF) {
+        return;
+    }
+
+    int_id = flex_int_mapping[dev] - 1;
+
+    NVIC_DisableIRQ(int_id);
+    LPC_GPIO_PIN_INT->CIENR = (1 << int_id);   // Disable rising edge or level interrupt
+    LPC_GPIO_PIN_INT->CIENF = (1 << int_id);    // Disable falling edge interrupt
+    LPC_GPIO_PIN_INT->RISE = (1 << int_id);    // Clear rising edge (sort of) flag
+    LPC_GPIO_PIN_INT->FALL = (1 << int_id);    // Clear falling edge (sort of) flag
+}
```

Comment:
Calling `gipo_irq_disable` currently omits interrupt edge configuration. Wouldn't it be sufficient to call `NVIC_DisableIRQ`? This way a consecutive call to `gpio_irq_enable` could resume with the init configuration.